smac_planner/src/a_star.cpp

```diff
@@ -364,6 +398,97 @@ bool AStarAlgorithm<NodeT>::isGoal(NodePtr & node)
   return node == getGoal();
 }
 
+template<>
+AStarAlgorithm<NodeSE2>::NodePtr AStarAlgorithm<NodeSE2>::getAnalyticPath(
+  const NodePtr & node,
+  const NodeGetter & node_getter)
+{
+  std::vector<std::pair<NodePtr, Coordinates>> possible_nodes;
+  const NodePtr & goal = getGoal();
+  ompl::base::ScopedState<> from(node->motion_table.state_space), to(
+    node->motion_table.state_space), s(node->motion_table.state_space);
+  const NodeSE2::Coordinates & node_coords = node->pose;
+  const NodeSE2::Coordinates & goal_coords = goal->pose;
+  from[0] = node_coords.x;
+  from[1] = node_coords.y;
+  from[2] = node_coords.theta * node->motion_table.bin_size;
+  to[0] = goal_coords.x;
+  to[1] = goal_coords.y;
+  to[2] = goal_coords.theta * node->motion_table.bin_size;
+
+  float d = node->motion_table.state_space->distance(from(), to());
+  NodePtr prev(node);
+  // A move of sqrt(2) is guaranteed to be in a new cell
+  constexpr float sqrt_2 = std::sqrt(2.);
+  unsigned int num_pts = std::floor(d / sqrt_2) + 1;
+  std::vector<double> reals;
+  // Pre-allocate
+  unsigned int index = 0;
+  NodePtr next(nullptr);
+  float angle = 0.0;
+  Coordinates proposed_coordinates;
+  // Don't generate the first point because we are already there!
+  // And the last point is the goal, so ignore it too!
+  for (unsigned int i = 1; i < num_pts; i++) {
+    node->motion_table.state_space->interpolate(from(), to(), (double)i / num_pts, s());
+    reals = s.reals();
+    angle = reals[2] / node->motion_table.bin_size;
+    while (angle >= node->motion_table.num_angle_quantization_float) {
```

Comment:
Is the output of interpolate ever non-normalize? See if these are ever entered, if not, remove them.

Reply:
Unfortunately it does need normalising at times.
Our angle space is effectively [0, 2pi), and OMPL is [-pi, pi)